modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -823,14 +846,17 @@ private CompletableFuture<Void> prepareTableResourcesAndLoadToZoneReplica(
         tables.put(tableId, table);
 
         // TODO: https://issues.apache.org/jira/browse/IGNITE-19913 Possible performance degradation.
-        return createPartsFut.thenAccept(ignore -> startedTables.put(tableId, table))
-                .whenComplete((v, th) -> {
-                    partitionReplicaLifecycleManager.unlockZoneForRead(zoneDescriptor.id(), stamp);
+        return createPartsFut.thenAccept(ignore -> {
+            startedTables.put(tableId, table);
 
-                    if (th == null) {
-                        addTableToZone(zoneDescriptor.id(), table);
-                    }
-                });
+            addTableToZone(zoneDescriptor.id(), table);
+        });
+    }
+
+    private void unlockZoneForRead(CatalogZoneDescriptor zoneDescriptor, CompletableFuture<Long> readLockAcquiryFuture) {
+        readLockAcquiryFuture.thenAccept(stamp -> {
```

Comment:
This makes me think: does the lock actually need the continuation executor? Shouldn't we just use `*Async` methods for some heavy chaining operations and for smaller ones use the default befavior? Do we really need an extra executor submission?